doc/src/core-transmutation.md

```diff
@@ -0,0 +1,143 @@
+Challenge: Preconditions for core transmuting methods
```

Comment:
Challenge ID is missing

Reply:
I'm including a number in the title. For example,
`Challenge 1: Preconditions for core transmuting methods`. Its not a blocker, just a nice to have.